### ISSUE TITLE
Plot option `maxTraces` is doing the opposite of what it's supposed to do

### ISSUE THREAD

**Issue by outofculture** [pyqtgraph/pyqtgraph#1763]
### Short description
`maxTraces` plot option is hiding the most recent `n` lines, rather than only showing them.

### Code to reproduce
```python
import pyqtgraph as pg
import numpy as np

p = pg.plot()
p.plotItem.ctrl.maxTracesCheck.setChecked(True)
p.plotItem.ctrl.maxTracesSpin.setValue(3)

for i in range(10):
    p.plot(np.random.normal(size=100) + i*10)
```

### Expected behavior
The plot should only show the 3 top-most lines.

### Real behavior
The plot shows only the 7 bottom-most lines.

### Tested environment(s)

 * PyQtGraph version: master
 * Qt Python binding: PyQt5 5.12.3
 * Python version: 3.7.10
 * NumPy version: 1.20.2
 * Operating system: linux, windows
 * Installation method: conda + git

### Additional Context
This behavior is the same when the option is controlled through the plot's menu interface. Hopefully I'll get around to fixing this presently.

**Comment by outofculture** [comment]
Oh, this behaves worse than I thought. _No_ lines are showing up in the plot with this option enabled. I tried to reproduce the effect of the user interface (which is what the bug describes), but huh. Oh well, the script still reproduces _something_.

Also, I just bisected to https://github.com/pyqtgraph/pyqtgraph/commit/b7b431de8d31c1d894ffc368ff5cd562fcc13678, with something else weird happening at https://github.com/pyqtgraph/pyqtgraph/commit/d77ad273c71870881c3fda3d2b49d3488f100425.
